evalml/pipelines/component_graph.py

```diff
@@ -514,8 +531,8 @@ def generate_order(cls, component_dict):
             return []
         digraph = nx.DiGraph()
         digraph.add_edges_from(edges)
-        if not nx.is_weakly_connected(digraph):
```

Comment:
I don't think this is necessary anymore given that edges have to be defined..? But open to discussion in case I've missed anything!

Reply:
We may have introduced a bug somewhere? I would expect the following graph to raise this error on `main` but it doesn't:

```python
from evalml.pipelines import ComponentGraph

cg = ComponentGraph({"Imputer": ["Imputer", "X", "y"],
                     "DateTime": ["DateTime Featurization Component", "X", "y"],
                     "Logistic Regression Classifier": ["Logistic Regression Classifier", "DateTime.x", "DateTime.y"]})
```

It's not completely connected!

![image](https://user-images.githubusercontent.com/41651716/127531458-3a889778-4375-42e6-8510-fcee5f31fa48.png)

Reply:
Yes, I think you're right--gonna look into this, thank you!

Reply:
@freddyaboulton Added the nodes to the DiGraph that we're using to determine connectivity so that if a node does not have an edge, it raises a `ValueError`, and added back the test in `test_init_bad_graphs`. Thanks for your sharp eyes 👀

Reply:
Thanks for digging into this!